frontend/src/metabase/components/ModalContent/ModalHeader.tsx

```diff
@@ -29,14 +30,33 @@ export const ModalHeader = ({
   const actionIconSize = fullPageModal ? 24 : 16;
 
   return (
-    <HeaderContainer className={className} data-testid="modal-header">
-      <HeaderTextContainer onClick={onBack}>
+    <Flex
+      gap="sm"
+      align="center"
+      p="xl"
+      className={cx(S.HeaderContainer, className)}
+      data-testid="modal-header"
+    >
+      <Flex
+        align="center"
+        w="100%"
+        className={cx(S.HeaderTextContainer, {
+          [S.clickable]: !!onBack,
+        })}
+        onClick={onBack}
+      >
         {onBack && <ModalHeaderBackIcon name="chevronleft" />}
 
-        <HeaderText textCentered={fullPageModal || centeredTitle}>
-          {children}
-        </HeaderText>
-      </HeaderTextContainer>
+        <Box
+          component="h2"
```

Comment:
Any reason why it's not `Title`?

```suggestion
        <Title
          order={2}
```

Reply:
can be, I didn't see such component 😄